uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -494,14 +506,18 @@ define([
       };
 
       /**
-       * Checks if given status is available to manager
+       * Checks if given status is available to manager. If manager applies leave
+       * on behalf of staff then cancelled is also removed from her list of available statuses.
        *
-       * @param {String} status
-       * @return {Boolean}
+       * @return {Function} that takes status as param and returns a boolean
        */
-      this.isStatusAvailableForManager = function (status) {
-        return !(status.name === 'admin_approved' || status.name === 'waiting_approval');
-      }
+      this.isStatusAvailableForManager = function () {
+        return function (status) {
```

Comment:
Its better to filter the Status's inside this.getStatuses and change name to getFilteredStatuses. Then we can avoid using filter and this returning of function which is really confusing.